### PR TITLE
Switch to use Rust Edition 2018

### PR DESCRIPTION
**Pull request by carllerche**

### CODE REVIEW

File: src/sys/unix/mod.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ pub use iovec::IoVec;
 
 use std::os::unix::io::FromRawFd;
 
-pub fn pipe() -> ::io::Result<(Io, Io)> {
+pub fn pipe() -> std::io::Result<(Io, Io)> {
```
Comment (stjepang) [inline src/sys/unix/mod.rs]: Shouldn't this be `crate::io::Result<(Io, Io)>`? It's the same re-exported type, but I mean just for consistency...

Comment (carllerche) [reply src/sys/unix/mod.rs]: I think the `io` mod w/ the re-exports should go away. I am just leaving that for a different PR.

File: src/sys/unix/tcp.rs [hunk 2]
```diff
@@ -1,19 +1,17 @@
+use crate::event::Evented;
+use crate::sys::unix::eventedfd::EventedFd;
+use crate::sys::unix::io::set_nonblock;
+use crate::sys::unix::uio::VecIo;
+use crate::{io, Interests, PollOpt, Registry, Token};
 use std::fmt;
-use std::io::{Read, Write};
-use std::net::{self, SocketAddr};
-use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
-use std::time::Duration;
```
Comment (stjepang) [inline src/sys/unix/tcp.rs]: unintentional newline?

File: test/test_tcp.rs [hunk 3]
```diff
@@ -8,10 +8,10 @@ use std::time::Duration;
 
 use net2::{self, TcpStreamExt};
 
+use crate::{TryRead, TryWrite};
```
Comment (stjepang) [inline test/test_tcp.rs]: The imports here are separated by newline, while everywhere else they're squished together.

**Pull request merged**